tests/test_components/test_IO.py

```diff
@@ -186,7 +190,7 @@ def test_validation_speed(tmp_path):
         _S = td.Simulation.from_file(path)
         time_validate = time() - time_start
         times_sec.append(time_validate)
-        assert set_datasets_to_none(S) == _S
+        set_datasets_to_none(S)._json_string == _S._json_string
```

Comment:
Missing the `assert` here?

Reply:
Thanks, good catch!

Also, I think the written out `(val1 is None) != (val2 is None)` is better, so will change that as well.